src/platform/APM32/pwm_output_apm32.c

```diff
@@ -144,65 +148,72 @@ static uint16_t pwmConvertToExternal(float motorValue)
 }
 
 static motorVTable_t motorPwmVTable = {
-    .postInit = motorPostInitNull,
+    .postInit = NULL,
     .enable = pwmEnableMotors,
     .disable = pwmDisableMotors,
     .isMotorEnabled = pwmIsMotorEnabled,
     .shutdown = pwmShutdownPulsesForAllMotors,
     .convertExternalToMotor = pwmConvertFromExternal,
     .convertMotorToExternal = pwmConvertToExternal,
+    .write = pwmWriteStandard,
+    .decodeTelemetry = NULL,
+    .updateComplete = pwmCompleteMotorUpdate,
+    .requestTelemetry = NULL,
+    .isMotorIdle = NULL,
 };
 
-motorDevice_t *motorPwmDevInit(const motorDevConfig_t *motorConfig, uint16_t idlePulse, uint8_t motorCount, bool useUnsyncedPwm)
+void motorPwmDevInit(motorDevice_t *device, const motorDevConfig_t *motorConfig, uint16_t idlePulse)
 {
     memset(motors, 0, sizeof(motors));
 
-    motorPwmDevice.vTable = motorPwmVTable;
+    if (!device || !motorConfig) {
+        return;
+    }
+
+    motorCount = device->count;
+    device->vTable = &motorPwmVTable;
 
+    useContinuousUpdate = motorConfig->useContinuousUpdate;
     float sMin = 0;
     float sLen = 0;
-    switch (motorConfig->motorPwmProtocol) {
+    switch (motorConfig->motorProtocol) {
     default:
-    case PWM_TYPE_ONESHOT125:
+    case MOTOR_PROTOCOL_ONESHOT125:
         sMin = 125e-6f;
         sLen = 125e-6f;
         break;
-    case PWM_TYPE_ONESHOT42:
+    case MOTOR_PROTOCOL_ONESHOT42:
         sMin = 42e-6f;
         sLen = 42e-6f;
         break;
-    case PWM_TYPE_MULTISHOT:
+    case MOTOR_PROTOCOL_MULTISHOT:
         sMin = 5e-6f;
         sLen = 20e-6f;
         break;
-    case PWM_TYPE_BRUSHED:
+    case MOTOR_PROTOCOL_BRUSHED:
         sMin = 0;
-        useUnsyncedPwm = true;
+        useContinuousUpdate = true;
         idlePulse = 0;
         break;
-    case PWM_TYPE_STANDARD:
+    case MOTOR_PROTOCOL_PWM :
         sMin = 1e-3f;
         sLen = 1e-3f;
-        useUnsyncedPwm = true;
+        useContinuousUpdate = true;
         idlePulse = 0;
         break;
     }
 
-    motorPwmDevice.vTable.write = pwmWriteStandard;
-    motorPwmDevice.vTable.decodeTelemetry = motorDecodeTelemetryNull;
-    motorPwmDevice.vTable.updateComplete = useUnsyncedPwm ? motorUpdateCompleteNull : pwmCompleteOneshotMotorUpdate;
-
     for (int motorIndex = 0; motorIndex < MAX_SUPPORTED_MOTORS && motorIndex < motorCount; motorIndex++) {
         const unsigned reorderedMotorIndex = motorConfig->motorOutputReordering[motorIndex];
         const ioTag_t tag = motorConfig->ioTags[reorderedMotorIndex];
         const timerHardware_t *timerHardware = timerAllocate(tag, OWNER_MOTOR, RESOURCE_INDEX(reorderedMotorIndex));
 
         if (timerHardware == NULL) {
             /* not enough motors initialised for the mixer or a break in the motors */
-            motorPwmDevice.vTable.write = &pwmWriteUnused;
-            motorPwmDevice.vTable.updateComplete = motorUpdateCompleteNull;
+            device->vTable = NULL;
```

Comment:
It would be cleaner to use return value to signal that initialization failed. Using vTable is quite confusing (and it is even worse when nil vtable is used after this function sets NULL)